sites/main-site/src/content/blog/2024/seqera-containers-part-2.mdx

```diff
@@ -0,0 +1,623 @@
+
+Renovate runs on a schedule and automatically updates software versions for us based on the
+specifications we've laid out in a [common config](https://github.com/nf-core/ops/blob/main/.github/renovate/default.json5).
+
+The magic starts with some nf-core automation to add renovate comments to the `environment.yml` file:
+
+```yml {5,7,9}
+channels:
+    - conda-forge
+    - bioconda
+dependencies:
+    # renovate: datasource=conda depName=bioconda/bwa
+    - bioconda::bwa=0.7.18
+    # renovate: datasource=conda depName=bioconda/samtools
+    - bioconda::samtools=1.20
+    # renovate: datasource=conda depName=bioconda/htslib
+    - bioconda::htslib=1.20.0
+```
+
+[These comments will be added](https://github.com/nf-core/modules/issues/6504) through
+[the batch module updates](https://github.com/nf-core/modules/issues/5828) happening this summer.
+Future modules will have these comments [added automatically and linted](https://github.com/nf-core/tools/issues/3184)
+by the nf-core/tools CLI.
+
+The comments allow some scary regexes to find the conda dependencies and their versions
+in nf-core/modules, and check if there's a new version available.
+If there is a new version available, the Renovate bot will create a PR bumping the version,
+which in turn will kick off the container creation GitHub Action.
+
+The process will be very similar to the diagram laid out above, however we can go a step further:
+if the new software versions have no effect on the results of the tests, the PR will be automatically merged:
+
+<div class="excalidraw">
+    <Fragment set:html={renovate_flow} alt="Container creation flow" />
+</div>
+
+So: if all tests pass, the pull request is automatically merged without human intervention.
+In case of test failures, the Renovate bot automatically requests a review from the appropriate
+module maintainer using the `CODEOWNERS` file.
+The maintainer then steps in to fix failing tests and request a final review before merging.
+
+This efficient process ensures that software dependencies stay current with minimal manual oversight,
+reducing noise and streamlining development workflows.
+This will hopefully be the end of the _"can I get a review on this version bump"_ requests in `#review-requests`!
+
+# Automation - Pipelines
+
+We now have nice, up to date software packaging for the shared nf-core/modules with minimal manual intervention.
+However, we need to propagate these changes to the pipelines that use these modules.
+There are two main areas that we need to address:
+
+## Building config files
+
+As [described above](#pipelines), pipelines will have a set of config files automatically generated
+that specify the container or conda environment for each process in the pipeline.
+
+Creation of these files will be triggered whenever installing, updating or removing a module, via the `nf-core` CLI.
+The config files will be completely regenerated each time, so there will never be any manual merging required.
```

Comment:
Then, if there are any local modules, their configurations will need to be in a separate config?

Reply:
Good point. I guess either:
1. The use `container` declaration as before and are picked up by `nextflow inspect` OR
2. They have a `meta.yml` file in their subdirectories which is picked up as with shared modules OR
3. They have their own Nextflow configuration outside of the auto-generated files

Local modules deserve their own section though 👍🏻